contracts/strategy/yearn/YearnStrategy.sol

```diff
@@ -156,22 +164,43 @@ contract YearnStrategy is IStrategy, AccessControl, Ownable, CustomErrors {
         if (amount == 0) revert StrategyAmountZero();
         uint256 uninvestedUnderlying = _getUnderlyingBalance();
 
-        if (amount > uninvestedUnderlying) {
-            uint256 _sharesToWithdraw = _underlyingToShares(
-                amount - uninvestedUnderlying
-            );
+        if (amount <= uninvestedUnderlying) {
+            underlying.safeTransfer(vault, amount);
 
-            if (_sharesToWithdraw > _getShares())
-                revert StrategyNotEnoughShares();
+            emit StrategyWithdrawn(amount);
 
-            // burn shares and withdraw required underlying to strategy
-            yVault.withdraw(_sharesToWithdraw, address(this), 1);
+            return;
         }
 
-        // transfer underlying to vault
-        underlying.safeTransfer(vault, amount);
+        uint256 _sharesToWithdraw = _underlyingToShares(
+            amount - uninvestedUnderlying
+        );
+
+        if (_sharesToWithdraw > _getShares()) revert StrategyNotEnoughShares();
+
+        // burn shares and withdraw required underlying to strategy
+        uint256 amountWithdrawn = yVault.withdraw(
+            _sharesToWithdraw,
+            address(this),
+            maxLossWithdrawParam
+        );
+
+        underlying.safeTransfer(vault, uninvestedUnderlying + amountWithdrawn);
+
+        emit StrategyWithdrawn(uninvestedUnderlying + amountWithdrawn);
```

Comment:
I was resolving a merge conflict with the changes when strategy can have uninvested underlying. We are actually making only one transfer since these are 2 exclusive code paths. The issue was that we had to emit the `StrategyWithdrawn` event accounting for the exact amount of underlying withdrawn from the Yearn vault. If you find this code confusing, I can try to refactor it so we have only 1 call to transfer the underlying to the vault instead of having it in 2 places for different code paths.

Reply:
[refactored](https://github.com/lindy-labs/sc_solidity-contracts/pull/154/commits/086de576fe601338d1365b417ea885fc8c5f7dab)